学习编写ebpf程序，注释文档

wynyibo opened the pull request:
1

wynyibo committed:
modified:   net_watcher/netwatcher.c
	new file:   "study/net_watcher\346\263\250\351\207\212.pdf"
	new file:   study/tcp_v4_time/Makefile
	new file:   study/tcp_v4_time/test
	new file:   study/tcp_v4_time/test.bpf.c
	new file:   study/tcp_v4_time/test.c
	new file:   study/tcp_v4_time/test.h

wynyibo committed:
deleted:    study/tcp_v4_time/.output/bpf/bpf.h
	deleted:    study/tcp_v4_time/.output/bpf/bpf_core_read.h
	deleted:    study/tcp_v4_time/.output/bpf/bpf_endian.h
	deleted:    study/tcp_v4_time/.output/bpf/bpf_helper_defs.h
	deleted:    study/tcp_v4_time/.output/bpf/bpf_helpers.h
	deleted:    study/tcp_v4_time/.output/bpf/bpf_tracing.h
	deleted:    study/tcp_v4_time/.output/bpf/btf.h
	deleted:    study/tcp_v4_time/.output/bpf/libbpf.h
	deleted:    study/tcp_v4_time/.output/bpf/libbpf_common.h
	deleted:    study/tcp_v4_time/.output/bpf/libbpf_legacy.h
	deleted:    study/tcp_v4_time/.output/bpf/libbpf_version.h
	deleted:    study/tcp_v4_time/.output/bpf/skel_internal.h
	deleted:    study/tcp_v4_time/.output/bpf/usdt.bpf.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/bpftool
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/btf.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/btf.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/btf_dumper.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/btf_dumper.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/common.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/common.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/disasm.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/disasm.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/gen.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/gen.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/json_writer.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/json_writer.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/bpf.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/bpf_core_read.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/bpf_endian.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/bpf_helper_defs.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/bpf_helpers.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/bpf_tracing.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/btf.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/hashmap.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/libbpf.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/libbpf_common.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/libbpf_internal.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/libbpf_legacy.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/libbpf_version.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/relo_core.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/skel_internal.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/include/bpf/usdt.bpf.h
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/libbpf.a
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/bpf.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/bpf_prog_linfo.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/btf.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/btf_dump.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/gen_loader.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/hashmap.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/libbpf.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/libbpf_errno.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/libbpf_probes.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/linker.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/netlink.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/nlattr.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/relo_core.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/ringbuf.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/str_error.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/strset.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/libbpf/staticobjs/usdt.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/main.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/main.o
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/xlated_dumper.d
	deleted:    study/tcp_v4_time/.output/bpftool/bootstrap/xlated_dumper.o
	deleted:    study/tcp_v4_time/.output/libbpf.a
	deleted:    study/tcp_v4_time/.output/libbpf/libbpf.a
	deleted:    study/tcp_v4_time/.output/libbpf/libbpf.pc
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/bpf.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/bpf_prog_linfo.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/btf.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/btf_dump.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/gen_loader.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/hashmap.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/libbpf.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/libbpf_errno.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/libbpf_probes.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/linker.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/netlink.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/nlattr.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/relo_core.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/ringbuf.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/str_error.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/strset.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/usdt.o
	deleted:    study/tcp_v4_time/.output/libbpf/staticobjs/zip.o
	deleted:    study/tcp_v4_time/.output/pkgconfig/libbpf.pc
 	deleted:    study/tcp_v4_time/.output/test.bpf.o
	deleted:    study/tcp_v4_time/.output/test.o
	deleted:    study/tcp_v4_time/.output/test.skel.h
	deleted:    study/tcp_v4_time/.output/test.tmp.bpf.o

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/data/connects.log:
```diff
@@ -0,0 +1,21 @@
+connection{pid="43488",sock="0xffff8eea45e9f500",src="192.168.60.136:49486",dst="60.191.81.66:443",is_server="0",sport="-",dport="-",cwnd="-",ssthresh="-",sk_buff="-",wmem_queued="-",backlog="0",maxbacklog="0",rwnd="62780",cwnd="10",ssthresh="2147483647",sndbuf="87040",wmem_queued="0",rx_bytes="4.582K",tx_bytes="2.193K",srtt="149024",duration="121826",total_retrans="0",fast_retrans="-",timeout_re ...
+connection{pid="43488",sock="0xffff8eea45e9d100",src="192.168.60.136:53318",dst="118.112.232.118:443",is_server="0",sport="-",dport="-",cwnd="-",ssthresh="-",sk_buff="-",wmem_queued="-",backlog="0",maxbacklog="0",rwnd="62780",cwnd="10",ssthresh="2147483647",sndbuf="87040",wmem_queued="0",rx_bytes="16.878K",tx_bytes="1.230K",srtt="80440",duration="104346",total_retrans="0",fast_retrans="-",timeout ...
+connection{pid="43488",sock="0xffff8eea45e9c800",src="192.168.60.136:38174",dst="118.112.232.117:443",is_server="0",sport="-",dport="-",cwnd="-",ssthresh="-",sk_buff="-",wmem_queued="-",backlog="0",maxbacklog="0",rwnd="62780",cwnd="10",ssthresh="2147483647",sndbuf="87040",wmem_queued="0",rx_bytes="45.081K",tx_bytes="1.580K",srtt="73005",duration="327682",total_retrans="0",fast_retrans="-",timeout ...
+connection{pid="43488",sock="0xffff8ee98669d100",src="192.168.60.136:43162",dst="58.222.29.185:80",is_server="0",sport="-",dport="-",cwnd="-",ssthresh="-",sk_buff="-",wmem_queued="-",backlog="0",maxbacklog="0",rwnd="64240",cwnd="10",ssthresh="2147483647",sndbuf="87040",wmem_queued="0",rx_bytes="0",tx_bytes="422",srtt="176430",duration="57812",total_retrans="0",fast_retrans="-",timeout_retrans="-" ...
```
日志文件不要提交

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/data/packets.log:
```diff
@@ -0,0 +1,975 @@
+packet{sock="0xffff8eea45e98000",seq="902936302",ack="663241396",mac_time="2",ip_time="6",tcp_time="14",http_info="-",rx="0"} 0
```
日志文件不要提交

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/netwatcher.bpf.c:
```diff
@@ -1160,4 +1160,41 @@ int BPF_KPROBE(tcp_enter_loss, struct sock *sk) {
 }
 
+//监控tcp窗口信息
+/*
```
这些代码是做什么用的呀

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/netwatcher.c:
```diff
@@ -38,7 +38,7 @@ static char packets_file_path[1024];
 static int sport = 0, dport = 0; // for filter
 static int all_conn = 0, err_packet = 0, extra_conn_info = 0, layer_time = 0,
-           http_info = 0, retrans_info = 0; // flag
+           http_info = 0, retrans_info = 0,dporttest=0; // flag
```
```suggestion
           http_info = 0, retrans_info = 0, dporttest=0; // flag
```

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/netwatcher.c:
```diff
@@ -159,10 +156,34 @@ static int print_conns(struct netwatcher_bpf *skel) {
         bytes_to_str(acked_bytes, d.bytes_acked);
+     /**  fprintf(file,
+                "connection{pid=\"%d\",sock=\"%p\",src=\"%s\",dst=\"%s\","
+                "is_server=\"%d\"",
```
不需要的代码可以删除

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/netwatcher.c:
```diff
@@ -353,18 +377,16 @@ int main(int argc, char **argv) {
         err = ring_buffer__poll(rb, 100 /* timeout, ms */);
 
-        if(http_info) {
+         if(http_info) {
```
```suggestion
        if(http_info) {
```

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/netwatcher.c:
```diff
@@ -320,26 +345,25 @@ int main(int argc, char **argv) {
+    }
+    //if(dporttest)
+    //{
+  //      printf("%-22s %-22s %-11s %-11s %-11s %-11s \n" ,
```
无用代码不要提交哈

helight commented on eBPF_Supermarket/Network_Subsystem/net_watcher/netwatcher.c:
```diff
@@ -234,32 +255,36 @@ static int print_packet(void *ctx, void *packet_info, size_t size) {
+                    "packet{sock=\"%p\",seq=\"%u\",ack=\"%u\","
+                    "mac_time=\"-\",ip_time=\"-\",tcp_time=\"-\",http_"
+                    "info=\"%s\",rx=\"%d\"} 0\n",
+                    pack_info->sock, pack_info->seq, pack_info->ack, http_data,
```
无用代码不要提交

helight commented on eBPF_Supermarket/Network_Subsystem/study/tcp_v4_time/test.c:
```diff
@@ -0,0 +1,162 @@
+#include <argp.h>
+#include <signal.h>
+#include <stdio.h>
+#include <time.h>
```
这个是个新功能吗？如果是，看是不是直接合并到 net_watcher中

helight commented:
study/net_watcher注释.pdf 这类学习的分析文档可以使用 md 格式提交到
https://github.com/linuxkerneltravel/lmp/tree/develop/eBPF_Documentation/content/study 这里，新建一个 net_watcher的目录
另外提交代码，log，二进制，测试代码都不要提交的

helight commented:
把 pr 和 issue 关联起来，你这个 pr 是针对那个 issue 的，关联进来，只要把 issue 地址贴到 pr 中，不过最好是把 pr 地址贴到 issue 中，这样就可以关联起来

helight commented:
不错继续努力提交，看了分析报告，还不错的

LinkinPF commented:
可以给本地配置一些code style的检查工具，然后bpf的中间生成物添加到.gitignore里面

wynyibo committed:
new file:   port_tracing/Makefile
	new file:   port_tracing/port_tracing.bpf.c
	new file:   port_tracing/port_tracing.c
	new file:   port_tracing/port_tracing.h
